### DISCUSSION TITLE
Build geometry using default_material

### DISCUSSION THREAD

**Discussion by dalarev** [NanoComp/meep#2662] [Show and tell]
Hi everybody, 

I wrote a "tutorial" script demonstrating how to use the `default_material` function to define geometry. Please find it [here](https://github.com/dalarev/MEEP_scripts/blob/main/test_default_material.ipynb).

I needed this capability for my own work, so I built a simple cylinder model to compare results using `default_material` with `GeometricObject`. 

We use a dft field monitor to compare field profiles resulting from both methods. They seem to match up almost exactly, which is a nice validation. 

Lessons learned:
- When defining the NumPy array to feed to `default_material`, make sure that your grid axes match those of the `sim`, or else the geometry will be scaled! This wasn't obvious to me - I assumed that meep would match spatial coordinates and insert the geometry appropriately. Perhaps this is spelled out in the documentation somewhere, but I was not aware of it.
- Comparing the `get_epsilon` data between the two methods yields some differences in the perimeter of the geometry. Perhaps this is due to slight grid and interpolation mismatches. This doesn't seem to impact the results.

Hope this is useful to someone!

![Screenshot from 2023-10-04 14-53-15](https://github.com/NanoComp/meep/assets/52722332/9ceccd71-459a-4b96-bcb8-c22a657c677d)

**Comment by dalarev** [comment]
This was in response to [this](https://github.com/NanoComp/meep/discussions/2660) thread in which I ask if one can Boolean AND `GeometricObject`s. I realized that eventually more complicated geometry would be needed, so the `default_material` method would be more versatile.

**Comment by Lotfi99999** [comment]
Thank you
